### ISSUE TITLE
Test against relevant python versions

### ISSUE THREAD

**Issue by tfardet** [InseeFrLab/pynsee#172]
I'd propose to stop testing against python 3.7, as it already reached [end of life](https://endoflife.date/python) and start using 3.11 instead.

I also wonder whether it is necessary to test against all intermediate versions and whether oldest and newest alone would not be sufficient (especially given how long the tests take to run). No strong opinion, though, just wondering whether it's worth it.

**Comment by hadrilec** [comment]
Indeed, great idea, let's discontinue the testing on 3.7, and test only on 3.8 and 3.11 then. The tests module, should be adapted though because some tests are run only on 3.7

**Comment by tfardet** [comment]
OK, will try to look at that by the end of the week! (unless someone else does it first)

**Comment by hadrilec** [comment]
I made this commit 860e27a3a058edf1c760171d3ae8f66275b78cdb in the PR #162, tests will be performed only on 3.8 and 3.11

**Comment by hadrilec** [comment]
after the merger of this PR #180, tests are made only on python 3.8 and 3.11

**Comment by tfardet** [comment]
Fixed by b6618715e1dc1dabe3e193ca2f1b8ce01d6682fe

**Comment by tfardet** [comment]
I thought we moved to test only against 3.8 and 3.11 but it seems it was reverted at some point later.
Was there a specific reason for the reversion or should we move back to 3.8 and 3.11 only?
Should we even move to 3.8 and 3.12 now that it's out?

More generally, should we define a policy for the python version we test? (I would propose to test against earliest version with security support and latest version only)
